### ISSUE TITLE
Interactions in HTML broken for some columns

### ISSUE THREAD

**Issue by cdeil** [pandas-profiling/pandas-profiling#663]
**Describe the bug**

For the simple and small example given below, the "Interactions" user interface is working for some columns, but not for others.

**To Reproduce**

_Data:_

Put this in a file `example.csv`:
```
datetime,sample_id,silo_id,comment,1Day,2Day,7Day,28Day,IntSet,Exp
2020-12-07 10:27:00,S3256/20,(blank),(blank),,,,,,
2020-12-03 09:35:00,S3220/20,20/2923,6,,,,,,
2020-12-03 09:35:00,S3244/20,20/2911,Silo 6,19.0,,,,,
2020-12-01 10:47:00,S3217/20,20/2891,(blank),19.7,32.8,,,,
2020-11-30 09:00:00,S3194/20,20/2876,6,22.7,35.1,,,145.0,0.5
2020-11-27 14:00:00,S3162/20,20/2862,6,18.6,30.8,47.2,,170.0,0.5
2020-11-26 12:49:00,S3161/20,20/2851,8,19.5,34.4,50.7,,155.0,0.5
2020-11-24 09:36:00,S3111/20,20/2830,6,20.3,32.8,49.6,,135.0,1.0
2020-11-23 13:56:00,S3109/20,20/2819,(blank),19.6,32.4,48.6,,140.0,1.0
2020-11-20 15:00:00,S3083/20,20/2805,6,17.5,29.6,46.7,,160.0,1.0
2020-11-19 11:00:00,S3082/20,20/2791,8,19.5,32.5,48.4,,140.0,1.5
2020-11-17 10:00:00,S3038/20,20/2764,6,19.0,32.0,48.5,,145.0,0.5
2020-11-16 15:30:00,S3007/20,20/2759,6,20.0,32.6,49.8,,160.0,0.5
2020-11-13 15:00:00,S2993/20,20/2743,6,20.9,33.4,47.7,,155.0,0.5
2020-11-12 11:00:00,S3010/20,20/2728,7,20.5,32.9,47.0,,150.0,0.5
2020-11-10 14:45:00,S2964/20,20/2705,6,18.5,30.9,46.8,,155.0,0.5
2020-11-09 09:20:00,S2950/20,20/2696,6,17.5,32.1,49.9,,145.0,
2020-11-06 13:00:00,S2909/20,20/2681,6,17.3,29.5,45.3,55.9,160.0,0.5
2020-11-05 10:35:00,S2911/20,20/2670,7,20.5,32.6,47.9,57.8,150.0,
2020-11-03 09:00:00,S2882/20,20/2646,5,19.1,33.0,49.0,59.6,165.0,
2020-11-02 09:25:00,S2870/20,20/2633,6,20.2,32.3,46.9,57.4,150.0,0.5
2020-10-30 14:30:00,S2855/20,20/2619,Silo 9 Train sample,21.0,33.1,47.6,58.3,150.0,
2020-10-29 09:01:00,S2837/20,20/2604,Silo 6 29/10 8:35,17.9,31.6,46.9,57.2,135.0,
2020-10-27 08:24:00,S2814/20,20/2584,6,17.9,32.0,47.4,58.2,170.0,0.5
2020-10-26 08:28:00,S2800/20,20/2574,6,18.6,32.2,48.7,61.3,175.0,
2020-10-23 13:20:00,S2802/20,20/2555,6,19.1,33.0,49.8,61.1,175.0,0.5
2020-10-22 08:45:00,S2764/20,20/2545,6,18.8,33.3,50.8,60.4,180.0,1.0
2020-10-20 14:35:00,S2727/20,20/2523,6,18.5,32.4,50.7,61.7,180.0,
2020-10-19 09:00:00,S2714/20,20/2510,6,18.3,32.8,50.1,62.4,170.0,1.0
2020-10-16 12:00:00,S2695/20,20/2492,8,19.3,32.3,48.1,58.8,170.0,0.5
```

_Code:_ Preferably, use this code format:
```python
import pandas as pd
from pandas_profiling import ProfileReport
df = pd.read_csv("example.csv")
profile = ProfileReport(df, progress_bar=False)
profile.to_file("example.html")
```
--> 

**Version information:**

* _Python version_: 3.8.6
* _Environment_: I first saw the issue from JupyterLab, but you can simply put the code in a Python file and run `python example.py` to generate the HTML.
* pandas: 1.0.5
* pandas_profiling: 2.9.0
* _`pip`_: see all versions https://gist.github.com/cdeil/8f2c59b57082ee5845357059df1cc235

**Additional context**

Here's a screenshot showing the bug in latest Chrome on MacOS.

Open the generated HTML, go to "interactions" and select the "28Day" column.
See how it's not selected or doesn't show up in the plot.

That column starts with a few `NaN` at the top, that might be the issue. (but further down in the column it does have normal values, selecting that "1Day" and "28Day" should work fine.

![Screenshot 2021-01-08 at 15 39 29](https://user-images.githubusercontent.com/852409/104027460-b6ac0280-51c7-11eb-8dd6-b3a300db021b.png)

**Comment by sbrugman** [comment]
The cause for this is a broken mapping from column to "clean name", which removes numbers (resulting in four times "day" in this your example). Will fix in the next release.

**Comment by sbrugman** [comment]
@cdeil I've just released v2.10.1 that should resolve your issue!

_Want me to continue to support this package? Please consider [sponsoring](https://github.com/sponsors/sbrugman)_

**Comment by cdeil** [comment]
Yes, the issue is resolved. Thank you!!!
